### PR TITLE
feat(as-5769): add route, controller, view for comments-question page

### PR DESCRIPTION
**Pull request by edpackard-work**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-5769

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/controllers/final-comment/comments-question.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+exports.postCommentsQuestion = async (req, res) => {
+	const {
+		body,
+		body: { errors = {}, errorSummary = [] }
+	} = req;
+
+	if (Object.keys(errors).length > 0) {
+		return res.render(VIEW.FINAL_COMMENT.COMMENTS_QUESTION, {
+			errors,
```
Comment (james-hill-kc) [inline packages/forms-web-app/src/controllers/final-comment/comments-question.js:27]: Would it make sense to pass in hasComments here as well? Not sure if we want to persist the user's choice in the event of errors

File: packages/forms-web-app/src/controllers/final-comment/comments-question.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			errorSummary
+		});
+	}
+
+	const hasComments = body['comments-question'] === 'yes';
+	req.session.finalComment = { hasComments: hasComments };
```
Comment (james-hill-kc) [inline packages/forms-web-app/src/controllers/final-comment/comments-question.js:33]: This should probably account for the possibility that the finalComment object could already exist (for example if the user has navigated back from a subsequent step)

File: packages/forms-web-app/src/views/final-comment/comments-question.njk [hunk 3]
```diff
@@ -0,0 +1,61 @@
+{# todo: change layout based on appeal type when component reused #}
+{% extends "layouts/full-appeal-banner/main.njk" %}
+
+{% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}
+{% from "govuk/components/radios/macro.njk" import govukRadios %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+
+{% set title = "Do you want to submit final comment? - Appeal a planning decision - GOV.UK" %}
```
Comment (james-hill-kc) [inline packages/forms-web-app/src/views/final-comment/comments-question.njk:8]: It feels like this should read "Do you want to submit a final comment" but I can see it matches the copy in the ticket. Perhaps one to check with design?

File: packages/forms-web-app/src/controllers/final-comment/comments-question.js (hunk 1 above)
Comment (edpackard-work) [reply packages/forms-web-app/src/controllers/final-comment/comments-question.js:27]: looking at other radio button pages in the full appeal journey, I don't see the user choice being persisted (the typical error would be that they haven't made a choice).

File: packages/forms-web-app/src/controllers/final-comment/comments-question.js (hunk 2 above)
Comment (edpackard-work) [reply packages/forms-web-app/src/controllers/final-comment/comments-question.js]: i've put in the 'dev only' workaround from your earlier ticket

**Review by james-hill-kc — APPROVED** [review]

**Pull request merged**